### PR TITLE
Use weak references to Apollo client in closures

### PR DESCRIPTION
**Pull request by gsabran**
This PR changes a few closures that were keeping a strong reference to the Apollo client. This is to make sure that as soon as the user of Apollo client releases its reference, all Apollo operations will stop.

### CODE REVIEW

File: Sources/Apollo/ApolloClient.swift [hunk 1]
```diff
@@ -206,7 +215,7 @@ private func wrapResultHandler<Data>(_ resultHandler: GraphQLResultHandler<Data>
 }
 
 private final class FetchQueryOperation<Query: GraphQLQuery>: AsynchronousOperation, Cancellable {
-  let client: ApolloClient
+  weak var client: ApolloClient?
```
Comment (designatednerd) [inline Sources/Apollo/ApolloClient.swift]: Why does this need to be `weak` here?

Comment (gsabran) [reply Sources/Apollo/ApolloClient.swift]: I think you're adding the operation to the client's queue
https://github.com/apollographql/apollo-ios/blob/509b2220c7a2e5dc5d0940b11361df0068ed4a0c/Sources/Apollo/ApolloClient.swift#L137-L138
so both the client and the operation will be retained here

Comment (designatednerd) [reply Sources/Apollo/ApolloClient.swift]: oh woof, good point.

**Comment by mitchellporter** [comment]
Anyone else who ends up here wondering why they're not getting any data back anymore... you're probably not holding onto a strong reference of the `ApolloClient` now that the retain cycle has been fixed.

**Comment by designatednerd** [comment]
Ha, good catch! I'll make a note of it on the release notes for `0.18.0`

**Pull request merged**